channel.go

```diff
@@ -553,6 +554,14 @@ func (ch *Channel) serve() {
 				OnCloseStateChange: ch.connectionCloseStateChange,
 				OnExchangeUpdated:  ch.exchangeUpdated,
 			}
+
+			if tlsConn, ok := netConn.(*tls.Conn); ok {
+				if err := tlsConn.Handshake(); err != nil {
+					netConn.Close()
+					return
```

Comment:
Why is this needed? From the [`tls.Conn.Handshake` docs](https://pkg.go.dev/crypto/tls#Conn.Handshake):
> Most uses of this package need not call Handshake explicitly: the first Read or Write will call it automatically.

Reply:
Yup, you are right TLS read/write will take care of it automatically. But when there is a TLS handshake error, read/write will fail with a handshake error. I did this to catch the TLS handshake error upfront, and in the following diff, add log and metrics around it.